tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/EngSys/CodeownersTools/CodeownersTools.cs

```
@@ -167,29 +167,6 @@ public async Task<string> UpdateCodeowners(
                 // Normalize service path
                 var normalizedPath = CodeownersHelper.NormalizePath(path);
 
-                // Validate service label (perform early so tests expecting exceptions aren't swallowed)
-                if (!string.IsNullOrEmpty(serviceLabel))
```

How does the below code behave if for some reason service label is still null? Should we be throwing here?

---

I don't think we need to throw an exception because the code already handles this scenario before these lines - we validate that at least one of either the service label or path must be provided.

---

@benbp Since service label is by default an empty string, is there any way that it could be null? The reason we don't just throw if service label is null is because we want to continue if we are given either the service label or path. Service label should just be passed into any methods below as a blank string, which the methods expect.

---

Sounds good. Yeah I should have said "empty or null" but regardless it's better to do just do the empty+null check in case the outer code ever changes (defensive coding, basically).